siliconcompiler/core.py

```diff
@@ -972,14 +971,14 @@ def _search(self, cfg, keypath, *args, field='value', mode='get', clobber=True):
                     elif val == False:
                         val = "false"
                 # checking if value has been set
-                if field not in cfg[param]:
-                    selval = cfg[param]['defvalue']
-                else:
-                    selval = cfg[param]['value']
+                # TODO: fix clobber!!
+                selval = cfg[param]['value']
                 # updating values
                 if cfg[param]['lock'] == "true":
                     self.logger.debug("Ignoring {mode}{} to [{keypath}]. Lock bit is set.")
                 elif (mode == 'set'):
+                    #print(keypath, "**", param, field, val, isinstance(val, list))
+                    #TODO: line below is broken, should check for field
                     if (selval in empty) | clobber:
```

Comment:
Can we resolve this TODO now, or do you think it will require more thought? I think line should just be:

`if (field != 'value') or (selval in empty) or clobber:`

Reply:
Please check post merge.